buildSrc/src/main/java/org/elasticsearch/gradle/docker/DockerSupportService.java

```diff
@@ -0,0 +1,375 @@
+                        if (lastResult.isSuccess() && composePath.isPresent()) {
+                            isComposeAvailable = runCommand(composePath.get(), "version").isSuccess();
+                        }
+                    }
+                }
+            }
+
+            boolean isAvailable = isVersionHighEnough && lastResult != null && lastResult.isSuccess();
+
+            this.dockerAvailability = new DockerAvailability(
+                isAvailable,
+                isComposeAvailable,
+                isVersionHighEnough,
+                dockerPath,
+                version,
+                lastResult
+            );
+        }
+
+        return this.dockerAvailability;
+    }
+
+    /**
+     * Given a list of tasks that requires Docker, check whether Docker is available, otherwise throw an exception.
+     *
+     * @throws GradleException if Docker is not available. The exception message gives the reason.
+     */
+    void assertDockerIsAvailable(List<String> tasks) {
+        DockerAvailability availability = getDockerAvailability();
+
+        // Docker installation is available and compatible
+        if (availability.isAvailable) {
+            return;
+        }
+
+        // No Docker binary was located
+        if (availability.path == null) {
+            final String message = String.format(
+                Locale.ROOT,
+                "Docker (checked [%s]) is required to run the following task%s: \n%s",
+                String.join(", ", DOCKER_BINARIES),
+                tasks.size() > 1 ? "s" : "",
+                String.join("\n", tasks)
+            );
+            throwDockerRequiredException(message);
+        }
+
+        // Docker binaries were located, but did not meet the minimum version requirement
+        if (availability.lastCommand.isSuccess() && availability.isVersionHighEnough == false) {
+            final String message = String.format(
+                Locale.ROOT,
+                "building Docker images requires minimum Docker version of %s due to use of multi-stage builds yet was [%s]",
+                MINIMUM_DOCKER_VERSION,
+                availability.version
+            );
+            throwDockerRequiredException(message);
+        }
+
+        // Some other problem, print the error
+        final String message = String.format(
+            Locale.ROOT,
+            "a problem occurred while using Docker from [%s]%s yet it is required to run the following task%s: \n%s\n"
+                + "the problem is that Docker exited with exit code [%d] with standard error output:\n%s",
+            availability.path,
+            availability.version == null ? "" : " v" + availability.version,
+            tasks.size() > 1 ? "s" : "",
+            String.join("\n", tasks),
+            availability.lastCommand.exitCode,
+            availability.lastCommand.stderr.trim()
+        );
+        throwDockerRequiredException(message);
+    }
+
+    private boolean isBlacklistedOs() {
```

Comment:
We should stay away from terms like "blacklist" and "whitelist".
```suggestion
    private boolean isExcludedOs() {
```

Reply:
Agreed. I've renamed this.